ebpfcore/ebpf_drv.c

```diff
@@ -333,10 +335,11 @@ _ebpf_driver_io_device_control(
 
 Done:
     if (status != STATUS_PENDING) {
-        if (async) {
+        if (wdf_request_ref_acquired) {
             ebpf_assert(status != STATUS_SUCCESS);
             // Async operation failed. Remove cancellable marker.
             (void)WdfRequestUnmarkCancelable(request);
+            WdfObjectDereference(request);
```

Comment:
Interesting - is this leak only if an async operation completes synchronously?

Reply:
If it completes synchronously or if it fails.